### PR TITLE
Bad Pixel Map Creation Function

### PR DESCRIPTION
**Pull request by maxwellmb**
## Describe your changes
The PR ports over the II&T bad pixel map creation implementation and includes a unit test. 

A weird quirk of this is that the step function to create a bad pixel map `create_bad_pixel_map`, needs to accept a dataset although it isn't used, since it only relies on dark and flat calibration files. 

## Type of change
- New feature (non-breaking change which adds functionality)

## Reference any relevant issues (don't forget the #)
This addresses Issue #94 

## Checklist before requesting a review
- [x] I have linted my code
- [x] I have verified that all unit tests pass in a clean environment and added new unit tests, as needed
- [x] I have verified that all docstrings are properly formatted and added new documentation, as needed

### CODE REVIEW

File: corgidrp/bad_pixel_calibration.py [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+
+def create_bad_pixel_map(dataset, master_dark, master_flat, dthresh = 5., ffrac = 0.8, fwidth = 32):
+    """
+    Compute a fixed bad pixel map for EXCAM from a master dark and flat.
+
+    Note that the master dark should not be a synthetic dark, since those are
+    (presumably) free of bad pixels. 
+
+    Args:
+        dataset (corgidrp.data.Dataset): A dataset that is ignored by this class.
```
Comment (semaphoreP) [inline corgidrp/bad_pixel_calibration.py:12]: define the data processing steps that need to be done before running this step;

File: corgidrp/bad_pixel_calibration.py [hunk 2]
```diff
@@ -0,0 +1,130 @@
+import numpy as np
+from corgidrp.data import BadPixelMap, Dataset
+
+def create_bad_pixel_map(dataset, master_dark, master_flat, dthresh = 5., ffrac = 0.8, fwidth = 32):
+    """
+    Compute a fixed bad pixel map for EXCAM from a master dark and flat.
+
+    Note that the master dark should not be a synthetic dark, since those are
```
Comment (semaphoreP) [inline corgidrp/bad_pixel_calibration.py:8]: check with Kevin on this

File: tests/test_badpixelmap.py [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+    
+    mocks.create_dark_calib_files(filedir = datadir)
+
+    ####### test data architecture
+    dark_filenames = glob.glob(os.path.join(datadir, "simcal_dark*.fits"))
+
+    dark_dataset = data.Dataset(dark_filenames)
+
+    ###### create dark
+    dark_frame = detector.create_dark_calib(dark_dataset)
```
Comment (semaphoreP) [inline tests/test_badpixelmap.py:34]: note that this is being removed in PR #105 in favor of `darks.build_trad_dark`. Is that ok here?

**Review by semaphoreP — COMMENTED** [review]
small comments, otherwise looks good to me!

File: corgidrp/bad_pixel_calibration.py (hunk 2 above)
Comment (maxwellmb) [reply corgidrp/bad_pixel_calibration.py:8]: Kevin says that it could be

File: tests/test_badpixelmap.py (hunk 3 above)
Comment (maxwellmb) [reply tests/test_badpixelmap.py]: ok, moved to using `build_trad_dark`

**Review by semaphoreP — APPROVED** [review]

**Pull request merged**